util/MultiplayerCommon.h

```diff
@@ -20,6 +21,132 @@ FO_COMMON_API extern const std::string SP_SAVE_FILE_EXTENSION;
 
 FO_COMMON_API extern const int ALL_EMPIRES;
 
+class GameRules;
+
+/////////////////////////////////////////////
+// Free Functions
+/////////////////////////////////////////////
+typedef void (*GameRulesFn)(GameRules&); ///< the function signature for functions that add Rules to the GameRules (void (GameRules&))
+
+/** adds \a function to a vector of pointers to functions that add Rules to
+  * the GameRules.  This function returns a boolean so that it can be used to
+  * declare a dummy static variable that causes \a function to be registered as
+  * a side effect (e.g. at file scope:
+  * "bool unused_bool = RegisterGameRules(&foo)"). */
+FO_COMMON_API bool RegisterGameRules(GameRulesFn function);
+
+/** returns the single instance of the GameRules class */
+FO_COMMON_API GameRules& GetGameRules();
+
+/** Database of values that control how the game mechanics function. */
+class FO_COMMON_API GameRules {
+public:
+    enum RuleType : int {
```

Comment:
it tracks the type of each rule... I thought the name and enum values were self explanatory...